crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -62,9 +64,77 @@ pub enum MethodInvocation {
 
 type RethnetStateType = Arc<RwLock<dyn SyncState<StateError>>>;
 
+struct _FilterCriteria {
```

Comment:
To avoid large, unwieldy files, I'd recommend adding this to `rethnet_rpc_server/filter.rs` file

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4197/commits/6aea14ce9e673962f3e2b3c4e0e840b588491757